sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedProcessorTest.java

```diff
@@ -332,10 +325,172 @@ public void staledLeaseAcquiring() {
+                    )
+                .then()
+            )
+            .subscribe();
+
+        // Wait for the feed processor to receive and process the first batch of documents and apply throughput change.
+        try {
+            Thread.sleep(4 * CHANGE_FEED_PROCESSOR_TIMEOUT);
+        } catch (InterruptedException e) {
+            log.error(e.getMessage());
+        }
+
+        // Loop through reading the current partition count until we get a split
+        //   This can take up to two minute or more.
+        String partitionKeyRangesPath = extractContainerSelfLink(createdFeedCollectionForSplit);
+        FeedOptions feedOptions = new FeedOptions();
+        feedOptions.setRequestContinuation(null);
+
+        AsyncDocumentClient contextClient = getContextClient(createdDatabase);
+        Flux.just(1).subscribeOn(Schedulers.elastic())
+            .flatMap( value -> {
+                log.warn("Reading current hroughput change.");
+                return contextClient.readPartitionKeyRanges(partitionKeyRangesPath, feedOptions);
+            })
+            .map(partitionKeyRangeFeedResponse -> {
+                int count = partitionKeyRangeFeedResponse.getResults().size();
+
+                if ( count < 2) {
+                    log.warn("Throughput change is pending.");
+                    throw new RuntimeException("Throughput change is not done.");
+                }
+                return count;
+            })
+            // this will timeout approximately after 3 minutes
+            .retry(40, throwable -> {
+                try {
+                    log.warn("Retrying...");
+                    Thread.sleep(CHANGE_FEED_PROCESSOR_TIMEOUT);
+                } catch (InterruptedException e) {
+                    log.error(e.getMessage());
+                }
+                return true;
+            })
+            .last().block();
+
+        assertThat(changeFeedProcessor.isStarted()).as("Change Feed Processor instance is running").isTrue();
+
+        // generate the second batch of documents
+        createReadFeedDocuments(createdFeedCollectionForSplit, FEED_COUNT);
+
+        // Wait for the feed processor to receive and process the second batch of documents.
+        waitToReceiveDocuments(2 * CHANGE_FEED_PROCESSOR_TIMEOUT, FEED_COUNT * 2);
+
+        changeFeedProcessor.stop().subscribeOn(Schedulers.elastic()).timeout(Duration.ofMillis(CHANGE_FEED_PROCESSOR_TIMEOUT)).subscribe();
+
+        for (CosmosItemProperties item : createdDocuments) {
+            assertThat(receivedDocuments.containsKey(item.getId())).as("Document with getId: " + item.getId()).isTrue();
+        }
+
+        // Wait for the feed processor to shutdown.
+        try {
+            Thread.sleep(CHANGE_FEED_PROCESSOR_TIMEOUT);
+        } catch (InterruptedException e) {
+            log.error(e.getMessage());
+        }
+        receivedDocuments.clear();
+    }
+
+    @Test(groups = { "simple" }, timeOut = CHANGE_FEED_PROCESSOR_TIMEOUT)
```

Comment:
"simple" test group is only invoked as integration test in the presence of a cosmosdb endpoint.
 
as this is unit test, not integration test, the group needs to be "unit". 

also the unit tests should go in a separate class, e.g.,  please move the test to  the new class ServiceItemLeaseTest.java

Reply:
Won't fix.

Reply:
> Won't fix.

why?